Assets/MixedRealityToolkit.SDK/Experimental/NoneNativeKeyboard/Scripts/UICollection.cs

```diff
@@ -0,0 +1,213 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.UI
+{
+    /// <summary>
+    /// This component represents and ordered collection of UI elements. You
+    /// can add to the UICollection by either dropping UI elements in the UI
+    /// element this component is attached to in the Editor or by calling the
+    /// AddItem(RectTransform) method. UI elements that are added to this
+    /// collection via the Editor will be automatically arranged when this
+    /// component executes.
+    /// To use this component attach it to a UI element (a GameObject with a
+    /// RectTransform component) such as an Image or Panel.
+    /// </summary>
+    [AddComponentMenu("UI/HoloUIKit/UICollection")]
+    [RequireComponent(typeof(RectTransform))]
+    [ExecuteInEditMode]
+    public class UICollection : MonoBehaviour
+    {
+        /// <summary>
+        /// The maximum width that the collection should expand to. If the value is -1.0 then it will use
+        /// the width specified by the RectTransform of this component's GameObject.
+        /// </summary>
+        [Experimental]
+        public float MaxWidth = -1.0f;
```

Comment:
Consider making these private and adding getters / setters. Not required for experimental code but would be for non-experimental code. Also perhaps consider using the existing GridObjectCollection (and perhaps adding missing functionality) over adding a new component for layout.